tests/ashley/test_course_locked.py

```diff
@@ -0,0 +1,765 @@
+        response = self.client.get(url_topic, follow=True)
+        url_topic_reply = f"{url_topic}post/create/"
+        self.assertNotContains(
+            response,
+            f'<a href="{url_topic_reply}" class="btn btn-primary btn-sm">'
+            '<i class="fa fa-comment fa-lg"></i>&nbsp;Post reply</a>',
+            html=True,
+        )
+        response = self.client.get(url_topic_reply, follow=True)
+        self.assertEqual(403, response.status_code)
+
+    def test_course_create_forum_after_lock(self):
+        """
+        Instructor locks the course, then a student connects.
+        After the course is locked, the instructor reconnects and
+        then creates a new forum whereas the course is supposed
+        to be locked. We check that the student still can't reply
+        or add topic in this new forum created after the locked.
+        """
+        forum = self._connects("instructor")
+
+        response = self.client.post(
+            f"/forum/admin/lock_course/{forum.id}/", follow=True
+        )
+        self.assertEqual(200, response.status_code)
+
+        # course is now marked as locked
+        context = LTIContext.objects.get(lti_id=self.context_id)
+        self.assertEqual(context.is_marked_locked, True)
+
+        # student connection
+        self._connects("student", uuid="NewOne", lis_person_sourcedid="enzo")
+
+        # a new Forum is created after the locking
+        forum2 = ForumFactory()
+        forum2 = self._connects("instructor", forum_uuid=forum2.lti_id)
+        # create a post in forum3
+        post = PostFactory(topic=TopicFactory(forum=forum2))
+
+        # student reconnects
+        self._connects("student", uuid="NewOne", lis_person_sourcedid="enzo")
+        # user can only read the forum
+        self.assertEqual(
+            ["can_see_forum", "can_read_forum"],
+            list(
+                GroupForumPermission.objects.filter(
+                    forum=forum2, group=context.get_base_group(), has_perm=True
+                ).values_list("permission__codename", flat=True)
+            ),
+        )
+
+        # info is present in forums view
+        response = self.client.get("/forum/")
+        self.assertContains(
+            response,
+            "This course has locked forums, non admin can only read the history.",
+        )
+        # info is present in the forum4 view
+        url_forum2 = f"/forum/forum/{forum2.slug}-{forum2.pk}/"
+        response = self.client.get(url_forum2)
+        self.assertContains(
+            response,
+            "This course has locked forums, non admin can only read the history.",
+        )
+        url_topic_create = f"{url_forum2}topic/create/"
+        # user is student but he can't create a new topic anymore
+        self.assertNotContains(
+            response,
+            f'<a href="{url_topic_create}" class="btn btn-primary btn-sm">'
+            '<i class="fa fa-comments fa-lg"></i>&nbsp;New topic</a>',
```

Comment:
This assertion is on top of the rest, there is self.assertEqual(403, response.status_code) trying to create topics just after,  this assert was really to show off.
I wouldn't know what more to test, IMO it's already overtested in the way that other tests are present in the files testing the permission, and django_machina is as well already tested on not being able to create post if we don't have the right.
What do you have in mind?

Reply:
ah I think I might get it, do you mean to just do the assert on url_topic_create for instance, to minimize, ok I think I understand...

Reply:
sorry you didn't get my point. If I try to test that something is not included in the page, it is weak to test that something as complicated as what you are looking is not in the text... because an extra space would cause the assertion to succeed when in fact the content is maybe in the page without the extra space...

So my point is that you should test that something minimal is not in the page. Like for example in your case:
```
self.assertNotContains(response, "New topic")
```